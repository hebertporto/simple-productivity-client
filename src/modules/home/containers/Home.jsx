import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

import SectionCentered from './../../shared/components/grid/SectionCentered'
import TableNovels from './../components/TableNovels'
import SideMenu from './../components/SideMenu'

import {
  signOut
} from './../../../redux/actions/auth/actions'

import { fetchAllNovels } from '../../../redux/actions/novel/actions'

import './../style/home.css'

class Home extends Component {
  componentDidMount = () => {
    this.props.fetchAllNovels()
  }

  render () {
    const { novels } = this.props
    return (
      <SectionCentered>
        <SideMenu />
        <TableNovels novels={novels} />
      </SectionCentered>
    )
  }
}

Home.propTypes = {
  logged: PropTypes.bool
}

const mapStateToProps = (state) => {
  return {
    logged: state.auth.logged,
    novels: state.novel.novels,
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    signOut: () => dispatch(signOut()),
    fetchAllNovels: () => dispatch(fetchAllNovels())
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(Home)
