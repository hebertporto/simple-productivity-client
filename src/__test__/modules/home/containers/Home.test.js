import React from 'react'
import { shallow } from 'enzyme'
import configureStore from 'redux-mock-store'

const middlewares = []
const mockStore = configureStore(middlewares)

import Home from '../../../../modules/home/containers/Home'

describe('Home Container', () => {
  const home = shallow(<Home {...props}/>, {context: { store: mockStore() }})
  it('should redenders properly', () => {
    expect(home).toMatchSnapshot()
  })
})