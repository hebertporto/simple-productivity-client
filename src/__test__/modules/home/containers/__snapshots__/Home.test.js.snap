// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Container should redenders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(Home) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "logged": false,
      "signOut": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "logged": false,
        "signOut": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
